app/controllers/users_controller.rb

```diff
@@ -2118,9 +2125,13 @@ def render_invite_error(message)
   end
 
   def serialize_found_users(users)
-    each_serializer =
-      SiteSetting.enable_user_status? ? FoundUserWithStatusSerializer : FoundUserSerializer
+    {
+      users:
+        ActiveModel::ArraySerializer.new(users, each_serializer: found_user_serializer).as_json,
+    }
+  end
 
-    { users: ActiveModel::ArraySerializer.new(users, each_serializer: each_serializer).as_json }
+  def found_user_serializer
+    SiteSetting.enable_user_status? ? FoundUserWithStatusSerializer : FoundUserSerializer
   end
```

Comment:
Im not sure "FoundUser" means a lot here? Can we return a not found user anyways?

---

Also Im not sure why we need to have the FoundUserWithStatusSerializer separate, and not just add the content in FoundUserSerializer as it's driven by a site setting, if we need finer control we can have it as an option of the serializer:

```ruby
ActiveModel::ArraySerializer.new(users, each_serializer: FoundUserSerializer, include_status: ...).as_json,
```